tests/integration/operator/operator_test.go

```diff
@@ -253,6 +255,19 @@ spec:
 	})
 })
 
+func getVersionFromIstiod() (string, error) {
+	output, err := kubectl.Exec(controlPlaneNamespace, "deploy/istiod", "pilot-discovery version")
+	if err != nil {
+		return "", fmt.Errorf("error getting version from istiod: %v", err)
+	}
+
+	matches := istiodVersionRegex.FindStringSubmatch(output)
+	if len(matches) > 0 && matches[1] != "" {
```

Comment:
will this work (ie, not crash with array out of bounds) if len==1?

Reply:
It wouldn't crash because FindStringSubmatch would either return nil or a slice with exactly 2 elements. But I've fixed the condition anyway.